### ISSUE TITLE
[Bug] Undo tree is polluted when dressing.nvim is used

### ISSUE THREAD

**Issue by Akianonymus** [smjonas/inc-rename.nvim#17]
I guess that's because it is actually modifying the buffer.

**Comment by smjonas** [comment]
You are right, I have no idea why this didn't occur to me while working on `dressing.nvim` integration. I currently don't know of a way to correctly restore the buffer state which is why I have opened this Neovim issue: https://github.com/neovim/neovim/issues/19856

**Comment by Akianonymus** [comment]
I was about to create the same issue when i got to know about command preview with custom commands, thank you for creating the issue.

Also, i recently saw a wilder.nvim gif where command like was stimulated in a popup, i don't how it actually works but maybe take a look at it.

https://github.com/gelguy/wilder.nvim#command-palette-experimental

**Comment by smjonas** [comment]
Thank you @Akianonymus, your suggestion was the missing piece of the puzzle!
